src/lib.rs

```diff
@@ -147,6 +150,38 @@ where
         }
     }
 
+    pub fn read_co2_ppm_and_temp_celcius(&mut self) -> nb::Result<(u16, f32), Error<E>> {
+        loop {
+            if let MhZ19CState::Idle = &mut self.state {
+                let uart = self.uart.take().unwrap();
+                self.state = MhZ19CState::ReadCo2AndTemperature(WriteAndReadResponse::new(
+                    uart,
+                    &*READ_CO2_AND_TEMPERATURE.as_ref(),
+                    [0u8; 9],
+                    9,
+                ));
+            }
+
+            self.poll()?;
+
+            let state = core::mem::take(&mut self.state);
+            if let MhZ19CState::ReadCo2AndTemperature(future) = state {
+                let (uart, buf) = future.into_return_value();
+                self.uart = Some(uart);
+                let frame = Frame::new(buf);
+                let data = Self::unpack_return_frame(Command::ReadCo2AndTemperature, &frame)
+                    .map_err(nb::Error::Other)?;
+
+                let co2_ppm = u16::from_be_bytes(data[2..4].try_into().unwrap());
+                let temp = f32::from(u16::from_be_bytes(data[..2].try_into().unwrap())) / 100.0;
+
+                return Ok((co2_ppm, temp));
```

Comment:
I would prefer to return a struct with named members instead of a tuple.

Reply:
I've fixed it